gnomad/utils/constraint.py

```diff
@@ -1310,30 +1314,32 @@ def add_gencode_annotations(
         chromosome=gencode_ht.interval.start.contig,
     )
 
-    # Obtain transcript annotations from GENCODE file.
-    gencode_transcript = (
-        gencode_ht.filter(gencode_ht.feature == "transcript")
-        .select("chromosome", "transcript_id", "level", "transcript_type")
+    # Obtain CDS annotations from GENCODE file and calculate CDS length and
+    # number of exons.
+    annotations_to_add = annotations + ("chromosome", "transcript_id", "length")
+
+    gencode_cds = (
+        gencode_ht.filter(gencode_ht.feature == "CDS")
+        .select(*annotations_to_add)
         .key_by("transcript_id")
         .drop("interval")
     )
 
-    # Obtain CDS annotations from GENCODE file and calculate CDS length and
-    # number of exons.
-    gencode_cds = gencode_ht.filter(gencode_ht.feature == "CDS").select(
-        "transcript_id", "length"
-    )
+    annotations_to_add.remove("length")
```

Comment:
doesn't this mean that length gets dropped?

Reply:
we only need it to calculate the total cds length but don't need it for grouping